### PR TITLE
Follow crystal 1.0.0 in shard.yml

### PR DESCRIPTION
**Pull request by kachick**
```console
$ crystal --version
Crystal 1.0.0 (2021-03-22)

LLVM: 9.0.1
Default target: x86_64-apple-macosx

$ crystal spec
.............

Finished in 1.39 milliseconds
13 examples, 0 failures, 0 errors, 0 pending
```

So I think this shard will work in crystal 1.0.0 ☺️ 

But user using crystal 1.0.0 and shards 0.14.1 raises an error when `--ignore-crystal-version` not given.

So this change might reduce the annoying error for shards users? 🤔

https://github.com/crystal-lang/shards/blob/addc26a3f22fee9fccb01cbb3e8878bef02d4295/docs/shard.yml.adoc

>When just a version number is used it has a different semantic compared to dependencies: x.y.z will be interpreted as ~> x.y, >= x.y.z (ie: >= x.y.z, < (x+1).0.0) honoring semver.

### CODE REVIEW

**Review by matthewmcgarvey — APPROVED** [review]
Thank you for this, it definitely helps!

**Review by jwoertink — CHANGES_REQUESTED** [review]
😆  That's funny. I was actually going through all of the repos today to add this same deal in. Thanks for the help!

This is for sure the right direction, but I also want to add in a CI matrix to account for testing against all of the versions. Can you update the ci.yml file to include these? You can see what I'm doing here https://github.com/luckyframework/pulsar/pull/11 The only difference is I'm not sure if 0.35.1 will work or not... 🤔  We can probably just use 0.36.1 and 1.0.0 in the matrix for now.

**Comment by kachick** [comment]
Sure! But... I didn't personally use Github Actions yet 😅 
So first I just copied and paste from https://github.com/luckyframework/pulsar/pull/11  => https://github.com/luckyframework/habitat/pull/70/commits/99dae281da728fafd68293ce3d6e2e07d135b868 😋 

It makes CI 🔴  

```
Run crystal bin/ameba.cr
  crystal bin/ameba.cr
  shell: sh -e {0}
Error: file 'bin/ameba.cr' does not exist
Error: Process completed with exit code 1.
```

Then updating ameba in https://github.com/luckyframework/habitat/pull/70/commits/8702d1243f09c98077150541de8f5fa8a5bfaf8a, https://github.com/luckyframework/habitat/pull/70/commits/20d27c49773a18a1edc8f9fec4adbd51e0db7d8c
And tried to use suggested config as https://github.com/luckyframework/habitat/pull/70/commits/0a54909c6423a38ebe6b9320afbcbf38d9251605

It still makes CI 🔴 (from another error message)

```
Run crystal-ameba/github-action@v0.2.9
  env:
    GITHUB_TOKEN: ***
/usr/bin/docker run --name e4157fa50bb8bf42a0a811376b0ce2fa49_2d5092 --label 5588e4 --workdir /github/workspace --rm -e GITHUB_TOKEN -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true --network github_network_c9a9eb949e89490d99f71a8ef794a7c8 -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/habitat/habitat":"/github/workspace" 5588e4:157fa50bb8bf42a0a811376b0ce2fa49
Unhandled exception: Response: Forbidden. Body: {"message":"Resource not accessible by integration","documentation_url":"https://docs.github.com/rest/reference/checks#create-a-check-run"} (Exception)
  from /app/src/ameba-github_action/github_client.cr:22:7 in 'handle_response'
  from /app/src/ameba-github_action/github_client.cr:14:7 in 'post'
  from /app/src/ameba-github_action/runner.cr:35:18 in 'create_check'
  from /app/src/ameba-github_action/runner.cr:17:7 in 'run'
  from /app/src/ameba-github_action.cr:4:1 in '__crystal_main'
  from /usr/share/crystal/src/crystal/main.cr:110:5 in 'main_user_code'
  from /usr/share/crystal/src/crystal/main.cr:96:7 in 'main'
  from /usr/share/crystal/src/crystal/main.cr:119:3 in 'main'
  from __libc_start_main
  from _start
  from ???
```

Hmm... I can't immediately fix it, I think. 🙇

File: shard.yml [hunk 1]
```diff
@@ -4,11 +4,11 @@ version: 0.4.6
...
-crystal: 0.36.1
+crystal: '>= 0.36.1, < 2.0.0'
 
 license: MIT
 
 development_dependencies:
   ameba:
     github: crystal-ameba/ameba
-    version: 0.13.4
+    version: 0.14.1
```
Comment (jwoertink) [inline shard.yml]: This needs to stay as 0.13.4. [See Ameba Issue](https://github.com/crystal-ameba/ameba/issues/208).

File: spec/habitat_spec.cr [hunk 2]
```diff
@@ -189,7 +189,7 @@ describe Habitat do
     FakeServer.settings.this_can_be_nil.should eq "not nil"
 
     FakeServer.settings.nilable_with_default.should eq "default"
-    FakeServer.configure { |settings| settings.nilable_with_default = nil }
+    FakeServer.configure(&.nilable_with_default=(nil))
```
Comment (jwoertink) [inline spec/habitat_spec.cr]: Reverting the Ameba update should also allow you to revert this change.

File: src/habitat.cr [hunk 3]
```diff
@@ -241,7 +241,7 @@ class Habitat
 
     class HabitatSettings
       {% if type_with_habitat.superclass && type_with_habitat.superclass.type_vars.size == 0 && type_with_habitat.superclass.constant(:HABITAT_SETTINGS) %}
-        {% for decl in type_with_habitat.superclass.constant(:HABITAT_SETTINGS).map { |setting| setting[:decl] } %}
+        {% for decl in type_with_habitat.superclass.constant(:HABITAT_SETTINGS).map(&.[:decl]) %}
```
Comment (jwoertink) [inline src/habitat.cr]: Same. Reverting Ameba should allow you to revert this change.

**Review by jwoertink — COMMENTED** [review]
I think maybe the pulsar CI might still need a little work. Take a look at the LuckyRouter CI https://github.com/luckyframework/lucky_router/blob/562ba0a4796460c35c2a166f09b94911f7799fb9/.github/workflows/ci.yml I think you can literally copy this one and it should run fine.

File: shard.yml (hunk 1 above)
Comment (kachick) [reply shard.yml]: Oh thank you for letting me know! 🙏

File: spec/habitat_spec.cr (hunk 2 above)
Comment (kachick) [reply spec/habitat_spec.cr]: Sure! Reverted in https://github.com/luckyframework/habitat/pull/70/commits/3d3d5827cfd075c27379be6dc5813627c5cf2354

File: src/habitat.cr (hunk 3 above)
Comment (kachick) [reply src/habitat.cr]: Sure! Reverted in https://github.com/luckyframework/habitat/pull/70/commits/3d3d5827cfd075c27379be6dc5813627c5cf2354

File: shard.yml (hunk 1 above)
Comment (kachick) [reply shard.yml]: Reverted in https://github.com/luckyframework/habitat/pull/70/commits/8f7ca12501859c58495541c0997ab659f7f00612!

**Comment by kachick** [comment]
@jwoertink Thank you for the useful reference! And thank you for letting me know about latest ameba issue!

CI looks 🟢 for now 😄 

This PR's commit history is so messy. Feel free to squash merging 🙏

**Review by jwoertink — APPROVED** [review]
Awesome! Thanks so much for doing this. Sorry for the little run-around there, but looks good now!

**Comment by kachick** [comment]
Thank you!
